app/src/androidTest/java/org/mozilla/fenix/helpers/idlingresource/NetworkConnectedIdlingResource.kt

```diff
@@ -0,0 +1,33 @@
+package org.mozilla.fenix.helpers.idlingresource
+
+import android.net.ConnectivityManager
+import androidx.core.content.getSystemService
+import androidx.test.espresso.IdlingResource
+import androidx.test.platform.app.InstrumentationRegistry
+import org.mozilla.fenix.ext.isOnline
+
+class NetworkConnectedIdlingResource : IdlingResource {
```

Comment:
nit: I like all classes to have kdocs

Reply:
To clarify, all classes except the test classes for classes under test (e.g. no need for kdoc on `NoNetworkAccessStartupTest` unless it has a particular need for it).

The reason I want class kdoc is that sometimes the class name doesn't make the class' purpose obvious to someone using the class. For example, `HomeActivity` seems fairly obvious to fenix developer but to a new dev, it begs many questions: if there is a `HomeActivity`, `HomeFragment`, and `BrowserFragment`, where is the `BrowserActivity`? Is this `Activity` the "main" activity of the app? Are there other significant activities or is this app a single-activity type of app?

Reply:
done